gpg_key.go

```diff
@@ -0,0 +1,188 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// Compile-time proof of interface implementation
+var _ GPGKeys = (*gpgKeys)(nil)
+
+// GPGKeys describes all the GPG key related methods that the Terraform Private Registry API supports.
+//
+// TFE API Docs: https://www.terraform.io/cloud-docs/api-docs/private-registry/gpg-keys
+type GPGKeys interface {
+	// Uploads a GPG Key to a private registry scoped with a namespace.
+	Create(ctx context.Context, options GPGKeyCreateOptions) (*GPGKey, error)
+
+	// Read a GPG key.
+	Read(ctx context.Context, keyID GPGKeyID) (*GPGKey, error)
+
+	// Update a GPG key.
+	Update(ctx context.Context, keyID GPGKeyID, options GPGKeyUpdateOptions) (*GPGKey, error)
+
+	// Delete a GPG key.
+	Delete(ctx context.Context, keyID GPGKeyID) error
+}
+
+// gpgKeys implements GPGKeys
+type gpgKeys struct {
+	client *Client
+}
+
+// GPGKey represents a signed GPG key for a TFC/E private provider.
+type GPGKey struct {
+	ID             string    `jsonapi:"primary,gpg-keys"`
+	AsciiArmor     string    `jsonapi:"attr,ascii-armor"`
+	CreatedAt      time.Time `jsonapi:"attr,created-at,iso8601"`
+	KeyID          string    `jsonapi:"attr,key-id"`
+	Namespace      string    `jsonapi:"attr,namespace"`
+	Source         string    `jsonapi:"attr,source"`
+	SourceURL      *string   `jsonapi:"attr,source-url"`
+	TrustSignature string    `jsonapi:"attr,trust-signature"`
+	UpdatedAt      time.Time `jsonapi:"attr,updated-at,iso8601"`
+}
+
+// GPGKeyID represents the set of identifiers used to fetch a GPG key.
+type GPGKeyID struct {
+	Namespace string
+	KeyID     string
+}
+
+// GPGKeyCreateOptions represents all the available options used to create a GPG key.
+type GPGKeyCreateOptions struct {
+	Type       string `jsonapi:"primary,gpg-keys"`
+	Namespace  string `jsonapi:"attr,namespace"`
+	AsciiArmor string `jsonapi:"attr,ascii-armor"`
+}
+
+// GPGKeyCreateOptions represents all the available options used to update a GPG key.
+type GPGKeyUpdateOptions struct {
+	Type      string `jsonapi:"primary,gpg-keys"`
+	Namespace string `jsonapi:"attr,namespace"`
+}
+
+func (s *gpgKeys) Create(ctx context.Context, options GPGKeyCreateOptions) (*GPGKey, error) {
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	req, err := s.client.newRequest("POST", "/api/registry/private/v2/gpg-keys", &options)
+	if err != nil {
+		return nil, err
+	}
+
+	g := &GPGKey{}
+	err = s.client.do(ctx, req, g)
+	if err != nil {
+		return nil, err
+	}
+
+	return g, nil
+}
+
+func (s *gpgKeys) Read(ctx context.Context, keyID GPGKeyID) (*GPGKey, error) {
+	if err := keyID.valid(); err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("/api/registry/private/v2/gpg-keys/%s/%s",
```

Comment:
I have mixed thoughts on not taking a parameter for the `registry-name` (which must be `private`). The API [docs](https://www.terraform.io/cloud-docs/api-docs/private-registry/gpg-keys#get-gpg-key) specify this as a parameter, and this is a parameter for the private registry provider APIs. What are your thoughts about this @sebasslash ?

Reply:
Hmm, well since it's always going to be `private`, I'd rather not expose it as part of the interface.